### PR TITLE
Fix partition resize on first boot

### PR DESCRIPTION
**Pull request by nautik1**
sgdisk correctly does the job of resizing the partition on 1st boot, however partx is not correctly informing the kernel of partition changes.
As a result, the link `/dev/disk/by-partlabel/gandiroot` does not exists and gandiroot filesystem (xvda2) is not resized.

It seems re-triggering udev events on the disk fixes the issue, creating the missing links in /dev/disk/.

### CODE REVIEW

**Comment by Corwind** [comment]
LGTM

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 1]
```diff
@@ -47,17 +48,28 @@ if which sgdisk >/dev/null && [ -b "$disk" ]; then
...
             err_code="$?"
-            loginfo "Error updating in-kernel partition layout for disk " \
+            msg="Error updating in-kernel partition layout for disk " \
                 "${disk}, return: ${err_code}"
+            echo "$msg" | logger -t "$SYSLOG_TARGET"
             exit "$err_code"
         fi
+        # partx is sometimes not enough, e.g. in ubuntu 20 it does not trigger
+        # /dev/disk/by-partlabel links creation
+        if ! partprobe "$disk" 2>&1 | logger -t "$SYSLOG_TARGET"; then
```
Comment (diconico07) [inline gandi-config/bootstrap.d/03-resize_main_disk:66]: you should add dependency to partprobe in package or check for partprobe presence

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 2]
```diff
@@ -76,7 +88,8 @@ if [ -e $device ]; then
         if [ -x "$resize2fs" ]; then
             if ! $resize2fs "$device" 2>&1 | logger -t "$SYSLOG_TARGET"; then
                 err_code="$?"
-                loginfo "Error resizing ext disk ${device}, return: ${err_code}"
+                msg="Error resizing ext disk ${device}, return: ${err_code}"
+                echo "$msg" | logger -t "$SYSLOG_TARGET"
```
Comment (diconico07) [inline gandi-config/bootstrap.d/03-resize_main_disk:92]:
```suggestion
                logger -t "$SYSLOG_TARGET" "$msg"
```

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 3]
```diff
@@ -87,7 +100,8 @@ if [ -e $device ]; then
         if [ -x "$xfscmd" ]; then
             if ! $xfscmd "$device" 2>&1 | logger -t "$SYSLOG_TARGET"; then
                 err_code="$?"
-                loginfo "Error resizing XFS disk ${device}, return: ${err_code}"
+                msg="Error resizing XFS disk ${device}, return: ${err_code}"
+                echo "$msg" | logger -t "$SYSLOG_TARGET"
```
Comment (diconico07) [inline gandi-config/bootstrap.d/03-resize_main_disk:104]:
```suggestion
                logger -t "$SYSLOG_TARGET" "$msg"
```

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 4]
```diff
@@ -47,17 +48,28 @@ if which sgdisk >/dev/null && [ -b "$disk" ]; then
...
                 "${disk}, return: ${err_code}"
+            echo "$msg" | logger -t "$SYSLOG_TARGET"
             exit "$err_code"
         fi
+        # partx is sometimes not enough, e.g. in ubuntu 20 it does not trigger
+        # /dev/disk/by-partlabel links creation
+        if ! partprobe "$disk" 2>&1 | logger -t "$SYSLOG_TARGET"; then
+            err_code="$?"
+            msg="Error informing kernel of partition change on ${disk}, return: ${err_code}"
+            echo "$msg" | logger -t "$SYSLOG_TARGET"
```
Comment (diconico07) [inline gandi-config/bootstrap.d/03-resize_main_disk:69]:
```suggestion
            logger -t "$SYSLOG_TARGET" "$msg"
```

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 5]
```diff
@@ -47,17 +48,28 @@ if which sgdisk >/dev/null && [ -b "$disk" ]; then
...
                 exit "$err_code"
             fi
         fi
         # Inform kernel of partition changes.
         if ! partx --update "$disk" 2>&1 | logger -t "$SYSLOG_TARGET"; then
             err_code="$?"
-            loginfo "Error updating in-kernel partition layout for disk " \
+            msg="Error updating in-kernel partition layout for disk " \
                 "${disk}, return: ${err_code}"
+            echo "$msg" | logger -t "$SYSLOG_TARGET"
```
Comment (diconico07) [inline gandi-config/bootstrap.d/03-resize_main_disk:61]:
```suggestion
            logger -t "$SYSLOG_TARGET" "$msg"
```

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 6]
```diff
@@ -47,17 +48,28 @@ if which sgdisk >/dev/null && [ -b "$disk" ]; then
         if [ "$disk_guid" = '00000000-0000-0000-0000-000000000000' ]; then
             if ! sgdisk --randomize-guids "$disk" 2>&1 | logger -t "$SYSLOG_TARGET"; then
                 err_code="$?"
-                loginfo "Error randomizing GPT UIDs on disk ${disk}, return: ${err_code}"
+                msg="Error randomizing GPT UIDs on disk ${disk}, return: ${err_code}"
+                echo "$msg" | logger -t "$SYSLOG_TARGET"
```
Comment (diconico07) [inline gandi-config/bootstrap.d/03-resize_main_disk:52]:
```suggestion
                logger -t "$SYSLOG_TARGET" "$msg"
```

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 7]
```diff
@@ -35,8 +35,9 @@ if which sgdisk >/dev/null && [ -b "$disk" ]; then
                         --typecode="${part_id}:${part_guid_code}" \
                         "$disk" 2>&1 | logger -t "$SYSLOG_TARGET"; then
                     err_code="$?"
-                    loginfo "Error resizing partition ${part_id} of GPT disk " \
+                    msg="Error resizing partition ${part_id} of GPT disk " \
                             "${disk}, return: ${err_code}"
+                    echo "$msg" | logger -t "$SYSLOG_TARGET"
```
Comment (diconico07) [inline gandi-config/bootstrap.d/03-resize_main_disk:40]:
```suggestion
                   logger -t "$SYSLOG_TARGET" "$msg"
```

File: gandi-config/bootstrap.d/03-resize_main_disk (hunk 1 above)
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: Added a check on command partprobe

File: gandi-config/bootstrap.d/03-resize_main_disk (hunk 2 above)
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: Added

File: gandi-config/bootstrap.d/03-resize_main_disk (hunk 3 above)
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: Added

File: gandi-config/bootstrap.d/03-resize_main_disk (hunk 6 above)
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: Added

File: gandi-config/bootstrap.d/03-resize_main_disk (hunk 5 above)
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: Added

File: gandi-config/bootstrap.d/03-resize_main_disk (hunk 7 above)
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: Added

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 8]
```diff
@@ -35,8 +35,9 @@ if which sgdisk >/dev/null && [ -b "$disk" ]; then
                         --typecode="${part_id}:${part_guid_code}" \
                         "$disk" 2>&1 | logger -t "$SYSLOG_TARGET"; then
                     err_code="$?"
-                    loginfo "Error resizing partition ${part_id} of GPT disk " \
+                    msg="Error resizing partition ${part_id} of GPT disk " \
                             "${disk}, return: ${err_code}"
```
[the remark this answers is not in the document]
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: Oops! Good catch :)

File: gandi-config/bootstrap.d/03-resize_main_disk [hunk 9]
```diff
@@ -47,17 +48,28 @@ if which sgdisk >/dev/null && [ -b "$disk" ]; then
...
+                echo "$msg" | logger -t "$SYSLOG_TARGET"
                 exit "$err_code"
             fi
         fi
         # Inform kernel of partition changes.
         if ! partx --update "$disk" 2>&1 | logger -t "$SYSLOG_TARGET"; then
             err_code="$?"
-            loginfo "Error updating in-kernel partition layout for disk " \
+            msg="Error updating in-kernel partition layout for disk " \
                 "${disk}, return: ${err_code}"
```
[the remark this answers is not in the document]
Comment (nautik1) [reply gandi-config/bootstrap.d/03-resize_main_disk]: thanks

**Comment by nautik1** [comment]
Actually the only interesting thing about partprobe was re-triggering udev events for the disk... so did that directly instead.

**Pull request merged**